### PR TITLE
[DO NOT MERGE] PR#6218 with backoff

### PR DESCRIPTION
**Pull request by vikin91**
This is an attempt to implement the PR https://github.com/stackrox/stackrox/pull/6218 using backoff.

### CODE REVIEW

File: pkg/retry/unconfirmed_message_handler.go [hunk 1]
```diff
@@ -2,120 +2,78 @@ package retry
...
+	s.resendFun = func() error {
+		go func() {
+			_ = send()
+		}()
+		select {
+		case <-s.gotAck.Done():
+			log.Infof("Got ack!")
+			s.gotAck.Reset()
+			return nil
+		case <-time.After(s.b.NextBackOff()):
```
Comment (jschnath) [inline pkg/retry/unconfirmed_message_handler.go:56]: If the Nack is expected to arrive after `s.b.NextBackOff()` time, can't we run into problems by checking if exactly that amount of time has passed before we received one? I would assume that we need some slack here to account for latency

File: compliance/collection/compliance/compliance.go [hunk 2]
```diff
@@ -108,19 +112,15 @@ func (c *Compliance) manageNodeScanLoop(ctx context.Context) <-chan *sensor.MsgF
...
 				log.Infof("Scanning node %q", nodeName)
 				msg, err := c.nodeScanner.ScanNode(ctx)
 				if err != nil {
 					log.Errorf("Error running node scan: %v", err)
 				} else {
-					c.umh.ObserveSending()
+					go c.umh.ExecOperation() // blocks until ack arrives
 					c.cache = msg.Clone()
-					nodeInventoriesC <- msg
+					// nodeInventoriesC <- msg
```
[the remark this answers is not in the document]
Comment (vikin91) [reply compliance/collection/compliance/compliance.go:123]: This PR is broken and does not work. I shared it to show you how far I got at the end of the timebox. So for sure that comment would need to be removed before we merge, but first we would need to make it work :)

File: pkg/retry/unconfirmed_message_handler_test.go [hunk 3]
```diff
@@ -19,89 +21,89 @@ type UnconfirmedMessageHandlerTestSuite struct {
...
-			nackAfter:       []time.Duration{},
+		"should attempt only once when ack arrives immediately": {
+			wait:                    500 * time.Millisecond,
+			expectedSendingAttempts: 1,
+			sendAfter:               []time.Duration{1 * time.Millisecond},
+			ackAfter:                []time.Duration{10 * time.Millisecond},
+			nackAfter:               []time.Duration{},
+		},
+		"should attempt 3 times within 5 seconds": {
+			wait:                    3100 * time.Millisecond,
```
[the remark this answers is not in the document]
Comment (vikin91) [reply pkg/retry/unconfirmed_message_handler_test.go:52]: Good catch! I corrected the test title in the original PR.

File: pkg/retry/unconfirmed_message_handler.go (hunk 1 above)
Comment (vikin91) [reply pkg/retry/unconfirmed_message_handler.go:56]: The question would be what is a goo latency? :) I went here with 0s extra latency, cause the ACK arriving 1ms later would not cause any extra waiting (another case of that switch would be triggered).

**Pull request closed without merging**